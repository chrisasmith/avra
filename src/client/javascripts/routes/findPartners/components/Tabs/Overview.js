import React from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { injectIntl, defineMessages } from 'react-intl'
import styles from '../styles/FindPartners.scss'
import * as findPartnersActions from '../../../../reducers/pages/findPartnersActions'
import { createLoadingSelector } from '../../../../reducers/selectors'
import GridCard, { CARD_TYPE } from '../../../../components/GridCards/GridCard'
import FilterBreadCrumb from '../../../../components/Filter/FilterBreadCrumb'

const propTypes = {
  data: PropTypes.object,
  filter: PropTypes.object,
  isFetching: PropTypes.bool,
  intl: PropTypes.object,
}

const messages = defineMessages({
  topProducts: {
    id: 'find_partners.tab.overview.top_products',
    defaultMessage: 'Top Products',
  },
  productCurrentlyOwn: {
    id: 'find_partners.tab.overview.currently_own',
    defaultMessage: 'Which of the following products do you currently own or have you bought recently?',
  },
  topServices: {
    id: 'find_partners.tab.overview.top_services',
    defaultMessage: 'Top Services',
  },
  serviceCurrentlyUse: {
    id: 'find_partners.tab.overview.service_currently_use',
    defaultMessage: 'Which of the following services do you currently use or have you used in the last 12 months? And which do you intend to use again in the next 12 months?',
  },
})

const OverviewTab = injectIntl(({ intl, filter, data, isFetching }) => {
  const leftColumn = [
    {
      type: CARD_TYPE.INDEX,
      dataKey: 'top_partners',
      cardData: {
        title: intl.formatMessage(messages.topProducts),
        description: intl.formatMessage(messages.productCurrentlyOwn),
      },
    },
  ]

  const rightColumn = [
    {
      type: CARD_TYPE.INDEX,
      dataKey: 'top_services',
      cardData: {
        title: intl.formatMessage(messages.topServices),
        description: intl.formatMessage(messages.serviceCurrentlyUse),
      },
    },
  ]

  return (
    <div className={styles.grid}>
      <div className={styles.breadCrumb}>
        <FilterBreadCrumb filter={filter} />
      </div>
      <div className="container">
        <div className="row row-grid">
          <div className="col-6">
            {leftColumn.map((card, idx) => (
              <GridCard key={idx} card={card} isFetching={isFetching} data={data} />
            ))}
          </div>
          <div className="col-6">
            {rightColumn.map((card, idx) => (
              <GridCard key={idx} card={card} isFetching={isFetching} data={data} />
            ))}
          </div>
        </div>
      </div>
    </div>
  )
})

function mapStateToProps(state) {
  const loadingSelector = createLoadingSelector([findPartnersActions.GET_FIND_PARTNERS_PAGE.SELF])
  const filter = state.filters.filter
  const data = state.pages.findPartners.data
  return { ...state, isFetching: loadingSelector(state), filter, data }
}

OverviewTab.propTypes = propTypes
export default connect(mapStateToProps)(OverviewTab)
