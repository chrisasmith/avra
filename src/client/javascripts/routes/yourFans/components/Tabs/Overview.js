import React from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { injectIntl, defineMessages } from 'react-intl'
import styles from '../styles/YourFans.scss'
import * as yourFansActions from '../../../../reducers/pages/yourFansActions'
import { createLoadingSelector } from '../../../../reducers/selectors'
import { MODULE_TYPE } from '../../../../components/GridCards/DashboardGraphModule'
import GridCard, { CARD_TYPE } from '../../../../components/GridCards/GridCard'
import FilterBreadCrumb from '../../../../components/Filter/FilterBreadCrumb'

const propTypes = {
  data: PropTypes.object,
  filter: PropTypes.object,
  isFetching: PropTypes.bool,
  intl: PropTypes.object,
}

const messages = defineMessages({
  fanPopulation: {
    id: 'your_fans.tab.overview.fan_population',
    defaultMessage: 'Fan Population',
  },
  fpByRegion: {
    id: 'your_fans.tab.overview.fp_by_region',
    defaultMessage: 'By Region',
  },
  fpRegion: {
    id: 'your_fans.tab.overview.fp_region',
    defaultMessage: 'Region',
  },
  useSocialPlatform: {
    id: 'your_fans.tab.overview.use_social_platform',
    defaultMessage: 'Do you use any of the following social media platforms?',
  },
  socialPlatform: {
    id: 'your_fans.tab.overview.social_platform',
    defaultMessage: 'Social media platforms',
  },
  favoriteTeam: {
    id: 'your_fans.tab.overview.favorite_team',
    defaultMessage: 'Favorite Team',
  },
  mediaBehavior: {
    id: 'your_fans.tab.overview.media_behavior',
    defaultMessage: 'Media Behavior',
  },
  howOftenDoYouUseSM: {
    id: 'your_fans.tab.overview.how_often_use_sm',
    defaultMessage: 'How often do you use the following media for news about sports?',
  },
  media: {
    id: 'your_fans.tab.overview.media',
    defaultMessage: 'Media',
  },
  fanEngagement: {
    id: 'your_fans.tab.overview.fan_engagement',
    defaultMessage: 'Fan Engagement',
  },
  watchAttendListen: {
    id: 'your_fans.tab.overview.watch_attend_listen',
    defaultMessage: 'Have you watched, attended or listened to events in the last 12 months?',
  },
})

const OverviewTab = injectIntl(({ intl, filter, data, isFetching }) => {
  const leftColumn = [
    {
      type: CARD_TYPE.GRAPH,
      dataKey: 'region',
      cardData: {
        showTotal: true,
        title: intl.formatMessage(messages.fanPopulation),
        description: intl.formatMessage(messages.fpByRegion),
        sideTitle: intl.formatMessage(messages.fpRegion),
        type: MODULE_TYPE.COUNT,
      },
    }, {
      type: CARD_TYPE.GRAPH,
      dataKey: 'social_media',
      cardData: {
        title: intl.formatMessage(messages.useSocialPlatform),
        sideTitle: intl.formatMessage(messages.socialPlatform),
        type: MODULE_TYPE.PERCENT,
      },
    },
  ]

  const rightColumn = [
    {
      type: CARD_TYPE.STRING,
      hide: !data || !data.property || !data.property.team,
      cardData: {
        title: intl.formatMessage(messages.favoriteTeam),
        value: data && data.property && data.property.team ? `${Math.round(data.property.team.favorite)}%` : null,
      },
    }, {
      type: CARD_TYPE.GRAPH,
      dataKey: 'social_media',
      cardData: {
        title: intl.formatMessage(messages.useSocialPlatform),
        sideTitle: intl.formatMessage(messages.socialPlatform),
        type: MODULE_TYPE.PERCENT,
      },
    }, {
      type: CARD_TYPE.GRAPH,
      dataKey: 'media_behavior',
      cardData: {
        title: intl.formatMessage(messages.mediaBehavior),
        description: intl.formatMessage(messages.howOftenDoYouUseSM),
        sideTitle: intl.formatMessage(messages.media),
        type: MODULE_TYPE.PERCENT,
      },
    }, {
      type: CARD_TYPE.GRAPH,
      dataKey: 'fan_engagement',
      cardData: {
        title: intl.formatMessage(messages.fanEngagement),
        description: intl.formatMessage(messages.watchAttendListen),
        type: MODULE_TYPE.PERCENT,
      },
    },
  ]

  return (
    <div className={styles.grid}>
      <div className={styles.breadCrumb}>
        <FilterBreadCrumb filter={filter} />
      </div>
      <div className="container">
        <div className="row row-grid">
          <div className="col-6">
            {leftColumn.map((card, idx) => (
              <GridCard
                key={idx}
                card={card}
                data={data}
                isFetching={isFetching}
                extra={{ propertyTitle: filter.property.name, benchmarkTitle: filter.benchmark.name }}
              />
            ))}
          </div>
          <div className="col-6">
            {rightColumn.map((card, idx) => (
              <GridCard
                key={idx}
                card={card}
                data={data}
                isFetching={isFetching}
                extra={{ propertyTitle: filter.property.name, benchmarkTitle: filter.benchmark.name }}
              />
            ))}
          </div>
        </div>
      </div>
    </div>
  )
})

function mapStateToProps(state) {
  const loadingSelector = createLoadingSelector([yourFansActions.GET_HOME_PAGE.SELF])
  const filter = state.filters.filter
  const data = state.pages.yourFans.data
  return { ...state, isFetching: loadingSelector(state), filter, data }
}

OverviewTab.propTypes = propTypes
export default connect(mapStateToProps)(OverviewTab)
