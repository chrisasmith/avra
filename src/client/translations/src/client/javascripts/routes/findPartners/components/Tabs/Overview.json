[
  {
    "id": "find_partners.tab.overview.top_products",
    "defaultMessage": "Top Products"
  },
  {
    "id": "find_partners.tab.overview.currently_own",
    "defaultMessage": "Which of the following products do you currently own or have you bought recently?"
  },
  {
    "id": "find_partners.tab.overview.top_services",
    "defaultMessage": "Top Services"
  },
  {
    "id": "find_partners.tab.overview.service_currently_use",
    "defaultMessage": "Which of the following services do you currently use or have you used in the last 12 months? And which do you intend to use again in the next 12 months?"
  }
]