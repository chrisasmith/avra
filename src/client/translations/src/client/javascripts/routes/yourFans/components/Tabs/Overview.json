[
  {
    "id": "your_fans.tab.overview.fan_population",
    "defaultMessage": "Fan Population"
  },
  {
    "id": "your_fans.tab.overview.fp_by_region",
    "defaultMessage": "By Region"
  },
  {
    "id": "your_fans.tab.overview.fp_region",
    "defaultMessage": "Region"
  },
  {
    "id": "your_fans.tab.overview.use_social_platform",
    "defaultMessage": "Do you use any of the following social media platforms?"
  },
  {
    "id": "your_fans.tab.overview.social_platform",
    "defaultMessage": "Social media platforms"
  },
  {
    "id": "your_fans.tab.overview.favorite_team",
    "defaultMessage": "Favorite Team"
  },
  {
    "id": "your_fans.tab.overview.media_behavior",
    "defaultMessage": "Media Behavior"
  },
  {
    "id": "your_fans.tab.overview.how_often_use_sm",
    "defaultMessage": "How often do you use the following media for news about sports?"
  },
  {
    "id": "your_fans.tab.overview.media",
    "defaultMessage": "Media"
  },
  {
    "id": "your_fans.tab.overview.fan_engagement",
    "defaultMessage": "Fan Engagement"
  },
  {
    "id": "your_fans.tab.overview.watch_attend_listen",
    "defaultMessage": "Have you watched, attended or listened to events in the last 12 months?"
  }
]