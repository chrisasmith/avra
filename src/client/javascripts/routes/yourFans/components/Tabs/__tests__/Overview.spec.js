import React from 'react'
import { shallow } from 'enzyme'
import configureMockStore from 'redux-mock-store'
import thunk from 'redux-thunk'
import { shallowToJson } from 'enzyme-to-json'
import Overview from '../Overview'

const mockStore = configureMockStore([thunk])

let store

const getTab = s => shallow(<Overview store={mockStore(s)} />).shallow()

describe('Overview Tab', () => {
  beforeEach(() => {
    store = {
      location: { },
      loading: {

      },
      error: {

      },
      filters: {
        filter: {
          property: {
            name: 'property',
          },
          benchmark: {
            name: 'benchmark',
          },
        },
      },
      pages: {
        yourFans: {
          data: null,
        },
      },
    }
  })

  it('should show', () => {
    const wrapper = getTab(store)

    expect(shallowToJson(wrapper)).toMatchSnapshot()
  })

  it('should show loading', () => {
    store.loading.GET_HOME_PAGE = true

    const wrapper = getTab(store)

    expect(shallowToJson(wrapper)).toMatchSnapshot()
  })
})
