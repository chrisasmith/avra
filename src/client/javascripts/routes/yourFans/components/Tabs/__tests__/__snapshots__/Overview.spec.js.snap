// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Overview Tab should show 1`] = `
<div
  className="grid"
>
  <div
    className="breadCrumb"
  >
    <FilterBreadCrumb
      filter={
        Object {
          "benchmark": Object {
            "name": "benchmark",
          },
          "property": Object {
            "name": "property",
          },
        }
      }
    />
  </div>
  <div
    className="container"
  >
    <div
      className="row row-grid"
    >
      <div
        className="col-6"
      >
        <GridCard
          card={
            Object {
              "cardData": Object {
                "description": "By Region",
                "showTotal": true,
                "sideTitle": "Regions",
                "title": "Fan Population",
                "type": "COUNT",
              },
              "dataKey": "region",
              "type": "GRAPH",
            }
          }
          data={null}
          extra={
            Object {
              "benchmarkTitle": "benchmark",
              "propertyTitle": "property",
            }
          }
          isFetching={false}
          key="0"
        />
        <GridCard
          card={
            Object {
              "cardData": Object {
                "sideTitle": "Social media platforms",
                "title": "Do you use any of the following social media platorms?",
                "type": "PERCENT",
              },
              "dataKey": "social_media",
              "type": "GRAPH",
            }
          }
          data={null}
          extra={
            Object {
              "benchmarkTitle": "benchmark",
              "propertyTitle": "property",
            }
          }
          isFetching={false}
          key="1"
        />
      </div>
      <div
        className="col-6"
      >
        <GridCard
          card={
            Object {
              "cardData": Object {
                "title": "Favorite Team",
                "value": null,
              },
              "hide": true,
              "type": "STRING",
            }
          }
          data={null}
          extra={
            Object {
              "benchmarkTitle": "benchmark",
              "propertyTitle": "property",
            }
          }
          isFetching={false}
          key="0"
        />
        <GridCard
          card={
            Object {
              "cardData": Object {
                "sideTitle": "Social media platforms",
                "title": "Do you use any of the following social media platorms?",
                "type": "PERCENT",
              },
              "dataKey": "social_media",
              "type": "GRAPH",
            }
          }
          data={null}
          extra={
            Object {
              "benchmarkTitle": "benchmark",
              "propertyTitle": "property",
            }
          }
          isFetching={false}
          key="1"
        />
        <GridCard
          card={
            Object {
              "cardData": Object {
                "description": "How often do you use the following media for news about sports?",
                "sideTitle": "Media",
                "title": "Media Behavior",
                "type": "PERCENT",
              },
              "dataKey": "media_behavior",
              "type": "GRAPH",
            }
          }
          data={null}
          extra={
            Object {
              "benchmarkTitle": "benchmark",
              "propertyTitle": "property",
            }
          }
          isFetching={false}
          key="2"
        />
        <GridCard
          card={
            Object {
              "cardData": Object {
                "description": "Have you watched, attended or listened to events in the last 12 months?",
                "title": "Fan Engagement",
                "type": "PERCENT",
              },
              "dataKey": "fan_engagement",
              "type": "GRAPH",
            }
          }
          data={null}
          extra={
            Object {
              "benchmarkTitle": "benchmark",
              "propertyTitle": "property",
            }
          }
          isFetching={false}
          key="3"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`Overview Tab should show loading 1`] = `
<div
  className="grid"
>
  <div
    className="breadCrumb"
  >
    <FilterBreadCrumb
      filter={
        Object {
          "benchmark": Object {
            "name": "benchmark",
          },
          "property": Object {
            "name": "property",
          },
        }
      }
    />
  </div>
  <div
    className="container"
  >
    <div
      className="row row-grid"
    >
      <div
        className="col-6"
      >
        <GridCard
          card={
            Object {
              "cardData": Object {
                "description": "By Region",
                "showTotal": true,
                "sideTitle": "Regions",
                "title": "Fan Population",
                "type": "COUNT",
              },
              "dataKey": "region",
              "type": "GRAPH",
            }
          }
          data={null}
          extra={
            Object {
              "benchmarkTitle": "benchmark",
              "propertyTitle": "property",
            }
          }
          isFetching={true}
          key="0"
        />
        <GridCard
          card={
            Object {
              "cardData": Object {
                "sideTitle": "Social media platforms",
                "title": "Do you use any of the following social media platorms?",
                "type": "PERCENT",
              },
              "dataKey": "social_media",
              "type": "GRAPH",
            }
          }
          data={null}
          extra={
            Object {
              "benchmarkTitle": "benchmark",
              "propertyTitle": "property",
            }
          }
          isFetching={true}
          key="1"
        />
      </div>
      <div
        className="col-6"
      >
        <GridCard
          card={
            Object {
              "cardData": Object {
                "title": "Favorite Team",
                "value": null,
              },
              "hide": true,
              "type": "STRING",
            }
          }
          data={null}
          extra={
            Object {
              "benchmarkTitle": "benchmark",
              "propertyTitle": "property",
            }
          }
          isFetching={true}
          key="0"
        />
        <GridCard
          card={
            Object {
              "cardData": Object {
                "sideTitle": "Social media platforms",
                "title": "Do you use any of the following social media platorms?",
                "type": "PERCENT",
              },
              "dataKey": "social_media",
              "type": "GRAPH",
            }
          }
          data={null}
          extra={
            Object {
              "benchmarkTitle": "benchmark",
              "propertyTitle": "property",
            }
          }
          isFetching={true}
          key="1"
        />
        <GridCard
          card={
            Object {
              "cardData": Object {
                "description": "How often do you use the following media for news about sports?",
                "sideTitle": "Media",
                "title": "Media Behavior",
                "type": "PERCENT",
              },
              "dataKey": "media_behavior",
              "type": "GRAPH",
            }
          }
          data={null}
          extra={
            Object {
              "benchmarkTitle": "benchmark",
              "propertyTitle": "property",
            }
          }
          isFetching={true}
          key="2"
        />
        <GridCard
          card={
            Object {
              "cardData": Object {
                "description": "Have you watched, attended or listened to events in the last 12 months?",
                "title": "Fan Engagement",
                "type": "PERCENT",
              },
              "dataKey": "fan_engagement",
              "type": "GRAPH",
            }
          }
          data={null}
          extra={
            Object {
              "benchmarkTitle": "benchmark",
              "propertyTitle": "property",
            }
          }
          isFetching={true}
          key="3"
        />
      </div>
    </div>
  </div>
</div>
`;
